code/parachain/frame/composable-traits/src/dex.rs

```diff
@@ -11,6 +11,44 @@ use serde::{Deserialize, Serialize};
 use sp_runtime::{BoundedBTreeMap, DispatchError, Permill};
 use sp_std::{collections::btree_map::BTreeMap, fmt::Debug, ops::Mul, vec::Vec};
 
+/// Specifies and amount together with the asset ID of the amount.
+#[derive(
```

Comment:
🥇 Happy birthday!!!

Default asset in Composable is invalid. You cannot have 0 asset. So zero amount also kind of also not type (zero amount is like no operation or None). So may be if it compiles `Default` can be removed. So not sure.